cmd/influxd/inspect/dump_tsi/dump_tsi.go

```diff
@@ -0,0 +1,513 @@
+package dump_tsi
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"text/tabwriter"
+
+	"github.com/influxdata/influxdb/v2/models"
+	"github.com/influxdata/influxdb/v2/tsdb"
+	"github.com/influxdata/influxdb/v2/tsdb/index/tsi1"
+	"github.com/spf13/cobra"
+)
+
+type args struct {
+	seriesFilePath string
+	paths          []string
+
+	showSeries         bool
+	showMeasurements   bool
+	showTagKeys        bool
+	showTagValues      bool
+	showTagValueSeries bool
+
+	measurementFilter *regexp.Regexp
+	tagKeyFilter      *regexp.Regexp
+	tagValueFilter    *regexp.Regexp
+
+	w *io.Writer
+}
+
+func NewDumpTSICommand() *cobra.Command {
+	var arguments args
+	var measurementFilter, tagKeyFilter, tagValueFilter string
+	cmd := &cobra.Command{
+		Use:   "dumptsi",
+		Short: "Dumps low-level details about tsi1 files.",
+		Long: `Usage: influx_inspect dumptsi [flags] path...
+    -series
+            Dump raw series data
+    -measurements
+            Dump raw measurement data
+    -tag-keys
+            Dump raw tag keys
+    -tag-values
+            Dump raw tag values
+    -tag-value-series
+            Dump raw series for each tag value
+    -measurement-filter REGEXP
+            Filters data by measurement regular expression
+    -series-file PATH
+            Path to the "_series" directory under the database data directory.
+            Required.
+    -tag-key-filter REGEXP
+            Filters data by tag key regular expression
+    -tag-value-filter REGEXP
+            Filters data by tag value regular expression
+One or more paths are required. Path must specify either a TSI index directory
+or it should specify one or more .tsi/.tsl files. If no flags are specified
+then summary stats are provided for each file.
+`,
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Parse filters.
+			if measurementFilter != "" {
+				re, err := regexp.Compile(measurementFilter)
+				if err != nil {
+					return fmt.Errorf("failed to parse regex: %w", err)
```

Comment:
```suggestion
					return fmt.Errorf("failed to parse regex %q: %w", measurementFilter, err)
```
And similar for the error messages for the tag-key and tag-value filters below